lib/Horde/Db/Adapter/Postgresql/Schema.php

```diff
@@ -1063,12 +1063,21 @@ public function resetPkSequence($table, $pk = null, $sequence = null)
                 $quotedTable = $this->quoteTableName($table);
                 $quotedPk = $this->quoteColumnName($pk);
 
-                $sql = sprintf('SELECT setval(%s, (SELECT COALESCE(MAX(%s) + (SELECT increment_by FROM %s), (SELECT min_value FROM %s)) FROM %s), false)',
+                if ($this->postgresqlVersion() >= 100000) {
+                    $sql = sprintf('SELECT setval(%s, (SELECT COALESCE(MAX(%s) + (SELECT increment_by FROM pg_sequences WHERE sequencename=%s), (SELECT min_value FROM pg_sequences WHERE sequencename=%s)) FROM %s), false)',
```

Comment:
Would this work on earlier Postgres versions too?

Reply:
It does NOT work on Postgres 8.4

Reply:
pg_sequences is supported by >10, the code is wrapped in a way it should work on on the same earlier versions it was running before.

server_version_num should work starting from 8.2 according to documentation
https://www.postgresql.org/docs/8.2/static/runtime-config-preset.html
but I don't have any 8.X around to test.

8.4 has been EOLed in July, 2014.

Last debian release with 8.4 was squeeze in 2011
Last Ubuntu was natty 11.04 in 2011
Last fedora was laughlin 14 in 2010

But RHEL 6.9 in 2017 was still coming out with 8.4.20 in 2017